src/components/videocard/VideoCard.jsx

```diff
@@ -16,7 +25,38 @@ export function VideoCard({ video }) {
     link,
   } = video
 
-  let location = useLocation()
+  const [openPlaylist, setOpenPlaylist] = useState(false)
+  const [createPlaylist, setCreatePlaylist] = useState(false)
+  const [playlistTitle, setPlaylistTitle] = useState(false)
+  const [playlistDescription, setPlaylistDescription] = useState(false)
+  const [playlistId, setPlaylistId] = useState('')
+  const [playlists, setPlaylists] = useState([])
+
+  const location = useLocation()
+  const { encodedToken } = useAuth()
+
+  useEffect(() => {
+    ;(async () => {
+      try {
+        const response = await axios.get(`/api/user/playlists`, {
+          headers: {
+            authorization: encodedToken,
+          },
+        })
+        setPlaylists(response.data.playlists)
+      } catch (error) {
+        console.log(error)
+      }
+    })()
+  }, [playlists])
+
+  useEffect(() => {
+    const path = location.pathname
+    if (path && path.includes('/playlist/')) {
+      setPlaylistId(path.substr(10))
```

Comment:
may be here you can also use useParams to get id from path.

Reply:
Tried but was not working